test/distributed/simple_test_api_signal_handling.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# Owner(s): ["oncall: r2p"]
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Simple test to verify the signal handling changes work.
+"""
+
+import os
+import sys
+from pathlib import Path
+
+from torch.testing._internal.common_utils import run_tests
+
+
+# Add the current directory to Python path
+sys.path.insert(0, str(Path(__file__).parent))
+
+
+def test_launch_config():
+    """Test that LaunchConfig accepts the new signals_to_handle parameter."""
+    try:
+        from torch.distributed.launcher.api import LaunchConfig
+
+        print("Testing LaunchConfig...")
+
+        # Test with default signals
+        config1 = LaunchConfig(min_nodes=1, max_nodes=1, nproc_per_node=1)
+        print(f"Default signals: {config1.signals_to_handle}")
+        assert config1.signals_to_handle == "SIGTERM,SIGINT,SIGHUP,SIGQUIT"
+        print("✓ Default signals test passed")
+
+        # Test with custom signals
+        config2 = LaunchConfig(
+            min_nodes=1,
+            max_nodes=1,
+            nproc_per_node=1,
+            signals_to_handle="SIGTERM,SIGUSR1,SIGUSR2",
+        )
+        print(f"Custom signals: {config2.signals_to_handle}")
+        assert config2.signals_to_handle == "SIGTERM,SIGUSR1,SIGUSR2"
+        print("✓ Custom signals test passed")
+
+        return True
+
+    except Exception as e:
+        print(f"✗ LaunchConfig test failed: {e}")
+        import traceback
+
+        traceback.print_exc()
+        return False
+
+
+def test_command_line_args():
+    """Test that the command line argument parsing works."""
+    try:
+        from torch.distributed.run import get_args_parser
+
+        print("\nTesting command line argument parsing...")
+
+        parser = get_args_parser()
+
+        # Test default behavior
+        args1 = parser.parse_args(["--standalone", "test_script.py"])
+        print(f"Default signals from args: {args1.signals_to_handle}")
+        assert args1.signals_to_handle == "SIGTERM,SIGINT,SIGHUP,SIGQUIT"
+        print("✓ Default command line args test passed")
+
+        # Test custom signals
+        args2 = parser.parse_args(
+            [
+                "--standalone",
+                "--signals-to-handle=SIGTERM,SIGUSR1,SIGUSR2",
+                "test_script.py",
+            ]
+        )
+        print(f"Custom signals from args: {args2.signals_to_handle}")
+        assert args2.signals_to_handle == "SIGTERM,SIGUSR1,SIGUSR2"
+        print("✓ Custom command line args test passed")
```

Comment:
I am not sure if you want to write a lot of print in the test...

Reply:
@fduwjj  resolved!